source/common/buffer/watermark_buffer.h

```diff
@@ -54,6 +54,7 @@ class WatermarkBuffer : public LibEventInstance {
   void postProcess() override { checkLowWatermark(); }
 
   void setWatermarks(uint32_t low_watermark, uint32_t high_watermark);
+  uint32_t high_watermark() const { return high_watermark_; }
```

Comment:
Huh, looks like!

source/common/router/config_impl.h:  uint32_t retryOn() const override { return retry_on_; }
source/common/router/config_impl.h:  uint32_t numRetries() const override { return num_retries_; }
source/common/http/access_log/request_info_impl.h:  const Optional<uint32_t>& responseCode() const override { return response_code_; }

I'll send a separate PR to clarify the style guide.